homeassistant/components/fitbit/sensor.py

```diff
@@ -627,86 +680,47 @@ def __init__(
     @property
     def icon(self) -> str | None:
         """Icon to use in the frontend, if any."""
-        if (
-            self.entity_description.key == "devices/battery"
-            and self.extra is not None
-            and (extra_battery := self.extra.get("battery")) is not None
-            and (battery_level := BATTERY_LEVELS.get(extra_battery)) is not None
-        ):
-            return icon_for_battery_level(battery_level=battery_level)
+        if self.entity_description.key == "devices/battery" and self.device is not None:
+            return icon_for_battery_level(battery_level=self.device.battery_level)
```

Comment:
Note to reviewer: This now just uses the battery level directly from the API rather than using an approximate.